### PR TITLE
CSR

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: src/client/crypto/rng.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/// Will be gone when Trussed gains RSA support.
+pub struct TrussedRng<'a, T>(pub &'a mut T);
+impl<T> trussed::service::RngCore for TrussedRng<'_, T>
+where
+    T: trussed::client::CryptoClient,
+{
+    fn next_u32(&mut self) -> u32 {
+        let mut buf = [0u8; 4];
+        self.fill_bytes(&mut buf[..]);
+        u32::from_be_bytes(buf)
```
Comment (krystian-hebel) [inline src/client/crypto/rng.rs]: Since this is random anyway, wouldn't `from_ne_bytes` be slightly faster in some cases?

**Comment by krystian-hebel** [comment]
@arturkow2000 is private key stored anywhere? Certificate doesn't make much sense if the private key is forgotten.

File: src/client/crypto/rng.rs (hunk 1 above)
Comment (arturkow2000) [reply src/client/crypto/rng.rs]: Fixed

**Comment by arturkow2000** [comment]
@krystian-hebel Forgot about that, anyway due to change to Ed25519 storing key would change so there is no need to add here.
I will add that as part of https://github.com/fobnail/fobnail/pull/35

**Review by krystian-hebel — APPROVED** [review]

**Comment by macpijan** [comment]
@krystian-hebel Fell free to merge when you approve stuff.

**Comment by krystian-hebel** [comment]
@macpijan ACK, I was waiting for CI to keep up.

**Pull request merged**